eos-updater-flatpak-installer/main.c

```diff
@@ -249,6 +249,37 @@ main (int    argc,
                          "Couldn’t apply some flatpak update actions for this boot: %s",
                          error->message);
 
+          break;
+        }
+      case EU_INSTALLER_MODE_PRINT:
+        {
+          g_autoptr(GHashTable) applied_flatpak_ref_actions = NULL;
+          g_autoptr(GHashTable) new_flatpak_ref_actions_to_apply = NULL;
+          g_autofree gchar *formatted_applied_flatpak_ref_actions = NULL;
+          g_autofree gchar *formatted_flatpak_ref_actions_to_apply = NULL;
+
+          applied_flatpak_ref_actions = eufi_determine_flatpak_ref_actions_to_check (NULL, &error);
+          if (applied_flatpak_ref_actions == NULL)
+            return fail (EXIT_FAILED,
+                         "Could not get information on which flatpak ref actions have been applied: %s",
+                         error->message);
+
+          formatted_applied_flatpak_ref_actions =
+            euu_format_all_flatpak_ref_actions ("All flatpak ref actions that should have been applied",
+                                                applied_flatpak_ref_actions);
+          g_print ("%s\n", formatted_applied_flatpak_ref_actions);
+
+          new_flatpak_ref_actions_to_apply = eufi_determine_flatpak_ref_actions_to_apply (NULL, &error);
+          if (new_flatpak_ref_actions_to_apply == NULL)
+            return fail (EXIT_FAILED,
+                         "Could not get information on which flatpak ref actions to apply: %s",
+                         error->message);
+
+          formatted_flatpak_ref_actions_to_apply =
+            euu_format_all_flatpak_ref_actions ("All flatpak ref actions that are not yet applied",
+                                                new_flatpak_ref_actions_to_apply);
+          g_print ("%s\n", formatted_flatpak_ref_actions_to_apply);
+
```

Comment:
Same as above, can you split this out into another commit?

Reply:
It strikes me that this probably repeats a lot of the existing debug messages in the tool. I wonder if it would make more sense to add a `--dry-run`option.